libs/common/include/helpers/EnumArray.h

```diff
@@ -44,6 +45,32 @@ struct EnumArray
     iterator end() noexcept { return elems + size(); }
     constexpr const_iterator end() const noexcept { return elems + size(); }
 
+    constexpr bool operator==(const EnumArray& rhs)
```

Comment:
before implementing everything by ourselfs, shouldnt we probably use something like https://github.com/Neargye/magic_enum instead? (header only, MIT license?)

Reply:
AFAIK the `EnumArray` is new and simple enough, we could replace the enum string output operators and such though and our own string-enum (`ENUM_WITH_STRING`) 
Yeah the magic_enum and the like is nice but they all use some brute-force approach to guess the enum names (it is mostly for enum<->string conversion) so it is likely heavy on compile-time where used and might break (it even says it uses "a compiler specific hack").
And this one is C++17, which is to recent for us(?)

Oh and we can't even use it in the place where we use `ENUM_WITH_STRING` (for `AddonId`) because for some reason we encode the author in the enumerator value and hence get values up to `0x00E0002` which is impossible for the magic_enum approach. So we'd need to convert that to regular enums without explicit values (which I wouldn't mind)

Reply:
ah okay havent thought about the c++17 part